libs/foundry-dev-tools/src/foundry_dev_tools/clients/compass.py

```diff
@@ -723,8 +731,24 @@ def api_search_projects(
         """
         if decorations is not None:
             decorations = get_decoration(decorations)
-        if page_size not in range(MINIMUM_PROJECTS_PAGE_SIZE, MAXIMUM_PROJECTS_PAGE_SIZE + 1):
-            page_size = DEFAULT_PROJECTS_PAGE_SIZE
+        if not MINIMUM_PROJECTS_PAGE_SIZE <= page_size <= MAXIMUM_PROJECTS_PAGE_SIZE:
+            page_size = (
+                MINIMUM_PROJECTS_PAGE_SIZE if page_size < MINIMUM_PROJECTS_PAGE_SIZE else MAXIMUM_PROJECTS_PAGE_SIZE
+            )
+            warnings.warn(f"Parameter `page_size` not within boundaries (1 <= N <= 500). Defaulting to {page_size}.")
+
+        valid_page_token = (
+            page_token is not None
+            and page_token.isdecimal()
+            and MINIMUM_PROJECTS_SEARCH_OFFSET <= int(page_token) <= MAXIMUM_PROJECTS_SEARCH_OFFSET
+        )
+
+        if not valid_page_token:
+            page_token = None
+            warnings.warn(
+                "Parameter `page_token` is invalid and will not be reset to default value. "
+                "For valid pattern see parameter description of this function."
+            )
```

Comment:
You are right, I saw my mistake.

I would even exlude the `page_token.isdecimal()` into a separate condition, because we would otherwise miss the check for non-numeric values resulting in an error.

```
        if page_token is not None:
            if page_token.isdecimal() is False:
                raise ValueError(
                    f"Parameter `page_token` ({page_token}) is expected to contain a number "
                    f"as the starting offset for the request. "
                    f"The search offset must be within the range from "
                    f"{MINIMUM_PROJECTS_SEARCH_OFFSET} to {MAXIMUM_PROJECTS_SEARCH_OFFSET}"
                )

            page_token_int = int(page_token)
            if page_token_int < MINIMUM_PROJECTS_SEARCH_OFFSET:
                raise ValueError(
                    f"Parameter `page_token` ({page_token_int}) is less than "
                    f"the minimum search offset ({MINIMUM_PROJECTS_SEARCH_OFFSET})"
                )
            elif page_token_int > MAXIMUM_PROJECTS_SEARCH_OFFSET:
                raise ValueError(
                    f"Parameter `page_token` ({page_token_int}) is greater than "
                    f"the maximum search offset ({MAXIMUM_PROJECTS_SEARCH_OFFSET})"
                )
```